### PR TITLE
🐛 Fix replay visual viewport resize support

### PR DESCRIPTION
**Pull request by amortemousque**
## Motivation

A regression has been introduced in the [PR](https://github.com/DataDog/browser-sdk/pull/2659): the replay visual viewport resize support has been removed.

## Changes

- Call trackVisualViewportResize in records
- Add tests in records.ts to test that all records types are collected

## Testing

- [ ] Local
- [ ] Staging
- [x] Unit
- [ ] End to end

---

I have gone over the [contributing](https://github.com/DataDog/browser-sdk/blob/main/CONTRIBUTING.md) documentation.

### CODE REVIEW

File: packages/rum/src/domain/record/record.spec.ts [hunk 1]
```diff
@@ -146,64 +151,6 @@ describe('record', () => {
...
-
-    it('adds a Focus record on blur', () => {
-      startRecording()
-      emitSpy.calls.reset()
-
-      window.dispatchEvent(createNewEvent('blur'))
-      expect(getEmittedRecords()[0].type).toBe(RecordType.Focus)
-    })
-
-    it('adds a Focus record on when taking a full snapshot', () => {
```
Comment (amortemousque) [inline packages/rum/src/domain/record/record.spec.ts:184]: Now covered here: https://github.com/DataDog/browser-sdk/blob/9fb112d3a65790e2a95b7c0d1c4d3df7485deeab/packages/rum/src/domain/record/startFullSnapshots.spec.ts#L55

File: packages/rum/src/domain/record/record.spec.ts [hunk 2]
```diff
@@ -146,64 +151,6 @@ describe('record', () => {
...
-      startRecording()
-      emitSpy.calls.reset()
-
-      // trigger full snapshot by starting a new view
-      newView()
-
-      expect(getEmittedRecords()[1].type).toBe(RecordType.Focus)
-    })
-
-    it('set has_focus to true if the document has the focus', () => {
```
Comment (amortemousque) [inline packages/rum/src/domain/record/record.spec.ts:194]: Already covered here: https://github.com/DataDog/browser-sdk/blob/9fb112d3a65790e2a95b7c0d1c4d3df7485deeab/packages/rum/src/domain/record/trackers/trackFocus.spec.ts#L26

File: packages/rum/src/domain/record/record.spec.ts [hunk 3]
```diff
@@ -146,64 +151,6 @@ describe('record', () => {
...
-      hasFocus = true
-      startRecording()
-      expect((getEmittedRecords()[1] as FocusRecord).data.has_focus).toBe(true)
-    })
-
-    it("set has_focus to false if the document doesn't have the focus", () => {
```
Comment (amortemousque) [inline packages/rum/src/domain/record/record.spec.ts:200]: Already covered here: https://github.com/DataDog/browser-sdk/blob/9fb112d3a65790e2a95b7c0d1c4d3df7485deeab/packages/rum/src/domain/record/trackers/trackFocus.spec.ts#L37

File: packages/rum/src/domain/record/record.spec.ts [hunk 4]
```diff
@@ -146,64 +151,6 @@ describe('record', () => {
...
-      expect(getEmittedRecords()[0].type).toBe(RecordType.Focus)
-    })
-
-    it('adds a Focus record on blur', () => {
-      startRecording()
-      emitSpy.calls.reset()
-
-      window.dispatchEvent(createNewEvent('blur'))
-      expect(getEmittedRecords()[0].type).toBe(RecordType.Focus)
-    })
```
Comment (amortemousque) [inline packages/rum/src/domain/record/record.spec.ts:182]: Basically covered here: https://github.com/DataDog/browser-sdk/blob/9fb112d3a65790e2a95b7c0d1c4d3df7485deeab/packages/rum/src/domain/record/startFullSnapshots.spec.ts#L55

File: packages/rum/src/domain/record/record.spec.ts [hunk 5]
```diff
@@ -422,6 +369,95 @@ describe('record', () => {
     })
   })
 
+  describe('should collect records', () => {
```
Comment (BenoitZugmeyer) [inline packages/rum/src/domain/record/record.spec.ts:372]: **👏 praise:** ‏looks great

**Review by BenoitZugmeyer — APPROVED** [review]
Good catch

**Review by thomas-lebeau — APPROVED** [review]

**Pull request merged**